assignments/closure.js

```diff
@@ -1,21 +1,60 @@
 // ==== Challenge 1: Write your own closure ====
 // Write a simple closure of your own creation.  Keep it simple!
+function grandparent() {
 
+  const grandpa = "Grandpa Richard I";
+  console.log(`My name is ${grandpa}`);
+  //debugger;
 
-/* STRETCH PROBLEMS, Do not attempt until you have completed all previous tasks for today's project files */
+  function father() {
+    const father = "Richard II";
+    console.log(`My name is ${father}, my father is ${grandpa}`);
+  //  debugger;
+
+    function grandson() {
+      const grandson = "Richard III"
+      console.log(`My name is ${grandson}, my father is ${father}, and my grandfather is ${grandpa}`);
+    //  debugger;
+    }// town
+
+    grandson();
+  }// city
+
+  father();
+}// kingdom
+
+console.log(grandparent());
```

Comment:
You don't actually have to `console.log` this function call - all functions without return values return `undefined`, which is what you'll see at the end of the logs for this line.